gemrb/core/System/VFS.cpp

```diff
@@ -256,11 +258,95 @@ void PathAppend(path_t& target, const path_t& name)
 	}
 }
 
+static const auto C_LOCALE = std::locale{"C.UTF-8"};
+
+// Returns true if both strings are equal after lower-casing
+static bool strlicmp(const char *a, const char *b) {
+	const unsigned char *itA = reinterpret_cast<const unsigned char*>(a),
+		*itB = reinterpret_cast<const unsigned char*>(b);
+
+	while (*itA != 0) {
+		unsigned char iA = *itA, iB = *itB;
+
+		// a is longer
+		if (iB == 0) {
+			return false;
+		}
+
+		// One is special, the other one isn't
+		if ((iA >= 128 && iB < 128) || (iA < 128 && iB >= 128)) {
+			return false;
+		}
+
+		// ASCII range
+		if (iA < 128) {
+			if (std::tolower(iA) != std::tolower(iB)) {
+				return false;
+			}
+
+			itA++, itB++;
+		} else {
+			// multibyte range (assuming no broken encoding)
+			auto getWidth = [](unsigned char c) -> uint8_t {
+				if ((c & 0xF0) == 0xF0) {
+					return 4;
+				} else if ((c & 0xE0) == 0xE0) {
+					return 3;
+				} else {
+					return 2;
+				}
+			};
+
+			// can't feed utf-8 bytes value into std::locale
+			auto getCodepoint = [](const unsigned char *it, uint8_t length) -> wchar_t {
+				switch (length) {
+					case 4: return ((0x7 | *it) << 24)
+						| ((0x3F | *(it + 1)) << 16)
+						| ((0x3F | *(it + 2)) << 8)
+						| (0x3F | *(it + 3));
+					case 3: return ((0xF | *it) << 16)
+						| ((0x3F | *(it + 1)) << 8)
+						| (0x3F | *(it + 2));
+					default: return ((0x1F | *it) << 8)
+						| (0x3F | *(it + 1));
+				}
+			};
+
+			uint8_t widthA = getWidth(iA), widthB = getWidth(iB);
+
+			wchar_t aL = std::tolower(getCodepoint(itA, widthA), C_LOCALE);
+			wchar_t bL = std::tolower(getCodepoint(itB, widthB), C_LOCALE);
+
+			// They have no lower case and thus remain different, or they are equal now
+			if (aL != bL) {
+				return false;
+			}
+
+			itA += widthA;
+			itB += widthB;
+		}
+	}
+
+	// b is longer
+	if (*itB != 0) {
+		return false;
+	}
+
+	// equal
+	return true;
+}
+
 static bool FindMatchInDir(const char* dir, char* item)
 {
+	bool expensiveCheck =
+		std::any_of(item, item + strlen(item) + 1, [](char c) { return c < 0; });
```

Comment:
if you change the parameter type then we can avoid `strlen`

Reply:
we have `MutableStringView`

Reply:
Ok will try, did not have this on screen yet.

Reply:
No more `strlen`  anything.